python/lsst/obs/decam/crosstalk.py

```diff
@@ -119,128 +118,46 @@ def prepCrosstalk(self, dataRef):
-        victim_exposure = dataRef.get()
-        det = victim_exposure.getDetector()
         visit = dataRef.dataId['visit']
-        ccdnum = det.getId()
-        ccdnum_str = '%02d' % ccdnum
         crosstalkSources = defaultdict(list)
+
         decamisr = IsrTask()  # needed for source_exposure overscan correction
-        for amp in det:
-            victim = ccdnum_str + amp.getName()
-            for source in sources[victim]:
-                source_ccdnum = int(source[:2])
-                try:
-                    source_exposure = butler.get('raw', dataId={'visit': visit, 'ccdnum': source_ccdnum})
-                except RuntimeError:
-                    self.log.warn('Cannot access source %d, SKIPPING IT' % source_ccdnum)
-                else:
-                    self.log.info('Correcting victim %s from crosstalk source %s' % (victim, source))
-                    if 'A' in source:
-                        amp_idx = 0
-                    elif 'B' in source:
-                        amp_idx = 1
-                    else:
-                        self.log.fatal('DECam source amp name does not contain A or B, cannot proceed')
-                    source_amp = source_exposure.getDetector()[amp_idx]
-                    # If doCrosstalkBeforeAssemble=True, then use getRawBBox().  Otherwise, getRawDataBBox().
-                    source_dataBBox = source_amp.getRawBBox()
-                    # Check to see if source overscan has been corrected, and if not, correct it first
-                    if not source_exposure.getMetadata().exists('OVERSCAN'):
-                        decamisr.overscanCorrection(source_exposure, source_amp)
-                        self.log.info('Correcting source %s overscan before using to correct crosstalk'
-                                      % source)
-                    source_image = source_exposure.getMaskedImage().getImage()
-
-                    source_data = source_image.Factory(source_image, source_dataBBox, deep=True)
-                    crosstalkSources[victim].append((source_data, coeffs[(victim, source)], amp_idx))
+        for sourceDet in crosstalk.interChip:
+            try:
+                source_exposure = butler.get('raw', dataId={'visit': visit, 'ccd': sourceDet})
+            except RuntimeError:
+                self.log.warn('Cannot access source %s, SKIPPING IT' % sourceDet)
+            else:
+                self.log.info('Correcting victim %s from crosstalk source %s' %
+                              (crosstalk._detectorName, sourceDet))
+
+                for amp in source_exposure.getDetector():
+                    decamisr.overscanCorrection(source_exposure, amp)
+
+                # Do not assemble here, as DECam CT is corrected before assembly.
+                crosstalkSources[sourceDet] = source_exposure
         return crosstalkSources
 
-    @pipeBase.timeMethod
-    def run(self, exposure, crosstalkSources=None):
```

Comment:
I'm going to assume this all moved to ip_isr and not panic.

Reply:
It's now inheriting the new `CrosstalkTask.run()`, and that has all the inter-chip logic now.